OpenRA.Game/Game.cs

```diff
@@ -956,9 +956,8 @@ public static void LoadMap(string launchMap)
 				Order.Command($"state {Session.ClientState.Ready}")
 			};
 
-			var path = Platform.ResolvePath(launchMap);
-			var map = ModData.MapCache.SingleOrDefault(m => m.Uid == launchMap) ??
-				ModData.MapCache.SingleOrDefault(m => m.Package.Name == path);
+			var package = new OpenRA.FileSystem.Folder(Platform.EngineDir).OpenPackage(launchMap, ModData.ModFiles);
+			var map = ModData.MapCache.SingleOrDefault(m => m.Uid == launchMap || m.Uid == Map.ComputeUID(package));
```

Comment:
Suggestion to only call `OpenPackage` if `m.Uid != launchMap`. Opening and parsing a package is 'expensive'.

`OpenPackage` may return null. You are not explicitly considering this. `ComputeUID` is then called with `null` which it does not support.

Reply:
This is only called once during startup so unless there's an elegant way to parse the package only after the UID was not found I wouldn't worry about changing this for performance reasons. An idea might be to check if `launchMap` is an UID or path (via regex or something like that).

Reply:
Or simpler, just run the old code first - if that fails try it as a filename.

Reply:
path. If it isn't a path then the parameter is invalid. Wanted to avoid regex since a hash could theoretically have any characters in it (ours is only numbers and letters, but that might change in the future). I could put it in a try/catch method though if you want it to be cleaner.

Reply:
I have updated the code to include a try/catch block for the NullReferenceException that occurs if no path can be found, and to exit with a message stating the path was invalid if no valid path was provided. This also works if the UID is invalid, since an invalid UID defaults to checking the path, and a UID will never resolve to a valid path, so it will also exit with the same message (have tested this).